core/tests/protractor_utils/ExplorationEditorPage.js

```diff
@@ -0,0 +1,846 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the exploration editor, for use in Protractor
+ * tests.
+ */
+
+var forms = require('./forms.js');
+var general = require('./general.js');
+var interactions = require('../../../extensions/interactions/protractor.js');
+var ruleTemplates = require(
+  '../../../extensions/interactions/rule_templates.json');
+
+var _NEW_STATE_OPTION = 'A New Card Called...';
+var _CURRENT_STATE_OPTION = '(try again)';
+
+var ExplorationEditorPage = function() {
+  /*
+   * Interactive elements
+   */
+  var addResponseDetails = element(
+    by.css('.protractor-test-add-response-details'));
+  var addResponseHeader = element(
+    by.css('.protractor-test-add-response-modal-header'));
+  var answerDescription = element(
+    by.css('.protractor-test-answer-description'));
+  var answerDescriptionFragment = element.all(
+    by.css('.protractor-test-answer-description-fragment'));
+  var commitMessageInput = element(
+    by.css('.protractor-test-commit-message-input'));
+  var multipleChoiceAnswerOptions = function (optionNum) {
+    return element(
+      by.cssContainingText('.protractor-test-html-select-option', optionNum));
+  };
+  var neutralElement = element(by.css('.protractor-test-neutral-element'));
+  var defaultResponseTab = element(
+    by.css('.protractor-test-default-response-tab'));
+  var editorWelcomeModal = element.all(
+    by.css('.protractor-test-welcome-modal'));
+  var editOutcomeDestBubble = element(
+    by.css('.protractor-test-dest-bubble'));
+  var editOutcomeDestStateInput = editOutcomeDestBubble.element(
+    by.css('.protractor-test-add-state-input'));
+  var editOutcomeDestAddExplorationId = element(
+    by.css('.protractor-test-add-refresher-exploration-id'));
+  var editOutcomeFeedback = element(
+    by.css('.protractor-test-edit-outcome-feedback'));
+  var editOutcomeDestDropdownOptions = function(targetOption) {
+    return element(by.cssContainingText('option', targetOption));
```

Comment:
This element can be found in `outcome_destination_editor_directive.html`. It is created by `ng-options` from a `<select>` element.